### DISCUSSION TITLE
SIZE_GRID_ID e SIZE_GRID_ROW_ID obrigatórios agora em produtos de categorias fashion

### DISCUSSION THREAD

**Discussion by dilneiss** [rodurma/forum-meli#15] [Geral]
Pessoal, recentemente o Mercado Livre fez uma alteração na categoria de fashion, onde agora ao realizar uma atualização, é obrigatório informar a id da tabela de tamanhos.
Estou tentando realizar aqui como exemplo, uma alteração nesse tênis: https://produto.mercadolivre.com.br/MLB-2652893543-tnis-feminino-torricella-novo-nova-tndencia-esportivo-_JM
Porém sempre me retorna erro informando o seguinte: `{
	"message": "Validation error",
	"error": "validation_error",
	"status": 400,
	"cause": [
		{
			"department": "structured-data",
			"cause_id": 2611,
			"type": "error",
			"code": "missing.fashion_grid.grid_row_id.values",
			"references": [
				"item.attributes"
			],
			"message": "Attribute [SIZE_GRID_ROW_ID] is missing"
		}
	]
}`

Então comecei a testar de todas as formas conforme a documentação em: https://developers.mercadolivre.com.br/pt_br/tabelas-de-medidas#Asociar-gu%C3%ADa-de-talle
Enviei dento do nó raiz.attributes o seguinte json:
`{
			"id": "SIZE_GRID_ID",
			"value_name": "210059"
		}`
Sem sucesso, retornou o erro: `{
	"message": "Validation error",
	"error": "validation_error",
	"status": 400,
	"cause": [
		{
			"department": "structured-data",
			"cause_id": 2611,
			"type": "error",
			"code": "missing.fashion_grid.grid_row_id.values",
			"references": [
				"item.attributes"
			],
			"message": "Attribute [SIZE_GRID_ROW_ID] is missing"
		}
	]
}`

Então coloquei também da seguinte forma:
`		{
			"id": "GENDER",
			"value_id": "339665"
		},
		{
			"id": "SIZE_GRID_ID",
			"value_name": "210059"
		},
		{
			"id": "SIZE_GRID_ROW_ID",
			"value_name": "210059:1"
		}`

Sem sucesso também, recebi o erro: `{
	"message": "Validation error",
	"error": "validation_error",
	"status": 400,
	"cause": [
		{
			"department": "structured-data",
			"cause_id": 2615,
			"type": "error",
			"code": "invalid.fashion_grid.size.values",
			"references": [
				"item.name"
			],
			"message": "Attribute [SIZE] is not valid"
		}
	]
}`

Alguém já passou por isso? como resolveu?

**Comment by gersonfs** [comment]
Estamos usando no nosso sistema, funciona assim:

`SIZE_GRID_ID` vc deve colocar nos attributes do **anúncio**
`SIZE_GRID_ROW_ID` é em outro lugar, nos array de `attributes`, para cada uma das **variations**

o exemplo está incompleto, mas é +- isso em php:

```
[
  'attributes' => [
    ['id' => 'SIZE_GRID_ID', 'value_name' => '210059']
  ],
  'variations' => [
    [
      'attributes' => [
         ['id' => 'SIZE_GRID_ROW_ID', 'value_name' => '210059:1']
       ]
    ]
  ]
]
```
